### ISSUE TITLE
Unclear error message - 

### ISSUE THREAD

**Issue by LukasWallrich** [RobbievanAert/puniform#3]
Thanks for this package. When using it with the data below, I get an unclear error `Error in if (get_LR_tau(prof.tau = 0, yi = yi, vi = vi, est = est, tau.est = tau.est,  :  missing value where TRUE/FALSE needed`. It works with each half of the data, so it seems to be particular to something about the summary statistics here - but I don't even know where to start troubleshooting this ...

``` r
nudge_dat <- structure(list(effect_size = c(0.22051011833961554, 0.34001817877879231, 
                                         0.41539999999999999, 1.0185414881072539, 0.68979999999999997, 
                                         0.35000000000000003, 0.24060000000000001, 0.23569999999999999, 
                                         -0.031145401361873777, 0.041133333333333334, 0.031320000000000001, 
                                         1.1392490937272901, -0.092549999999999966, 0.53570000000000007, 
                                         0.22340157133496508, 0.23580000000000004, 0.060073906439910452, 
                                         0.26179999999999998, 0.21374854394529136, 1.0707478948206963, 
                                         0.047458302630698869, 0.45900000000000002, 0.2557936092392582, 
                                         0.99787988591829269, 0.45546519510931049, -0.017600000000000001, 
                                         0.49743436979096206, 0.39259999999999995, 0.63484763441531888, 
                                         0.3054, -0.04084466930578045, 0.16708187782011683, 0.10844904733366972, 
                                         -0.098799999999999999, 0.25054440251601123, -0.012299999999999998, 
                                         0.029428040129174011, 0.015299999999999999, 0.71389999999999998, 
                                         0.24670000000000003, 0.2475, -0.11438942447501378, 0.16339999999999999, 
                                         -0.027099999999999999, 0.30008359081861541, 0.13496495928486715, 
                                         0.059600000000000007, 0.059199999999999996, 0.24313046556970327, 
                                         0.40222991419066473, 0.55555182634617806, 0.57326627781684802, 
                                         0.088916690029157655, 1.4206920489077333, 0.54481250290616601, 
                                         0.35917836444486151, 0.45732087856976339, 0.17571692750662071, 
                                         0.28570000000000001, 0.4587, 0.62470000000000003, 0.47805778117216774, 
                                         0.48976529080613235, 2.0508741501730001, 0.44729999999999998, 
                                         0.72855127476241321, -0.015600000000000001, 0.17289999999999997, 
                                         0.47370000000000001, -0.0058680171171112978, 0.048827116852389066, 
                                         -0.018200000000000001, -0.01500305404801271, -0.091269023654530837, 
                                         0.050900000000000001, 0.71090000000000009, 0.6321, -0.10969999999999999, 
                                         0.86529999999999996, 0.22932316331719715, 0.24779999999999999, 
                                         0.311, 1.5448, 0.085800000000000001, 0.0032666666666666725, 0.69716713725611601, 
                                         0.58407201842300849, 0.85519999999999996, 0.52243280779080215, 
                                         1.0483481809951818, 0.31119999999999998, 0.067100000000000007, 
                                         0.31496309114765708, 0.55500000000000005, 0.55678027854751433, 
                                         0.21300000000000002, 0.94892661508570475, 0.72256434047628415, 
                                         0.062874191335311372, 0.48113619972140925, 0.40110000000000001, 
                                         1.3282159699379554, 0.54990000000000006, -0.2731887964320508, 
                                         0.80740000000000001, 0.27909999999999996, 0.33630662776175929, 
                                         0.26960000000000001, 0.079699999999999993, 1.6796907636970135, 
                                         0.11521936181719904, 0.095431224475845192, 0.49709653335841131, 
                                         0.099300979609447268, 1.0609, 0.64870000000000005, 0.31235065889660107, 
                                         0.075499999999999998, 0.67522857836888694, 0.10359053272043177, 
                                         1.4775, 0.50949999999999995, 0.32250000000000001, 0.2557757115144898, 
                                         0.60909999999999997, 0.07427479489700288, 0.21510645829746677, 
                                         -0.021000000000000001, 0.26343107397083237, 0.53400000000000003, 
                                         0.7643467254245826, 0.34618639293395004, 0.065930315091306368, 
                                         0.51400000000000001, 0.090200000000000002, 0.32140000000000002, 
                                         0.39604346453385786, -0.0258, 0.40539999999999998, 0.53576059838196177, 
                                         0.48718221006101986, 1.4046071021835349, 0.1704, 0.1195226565742139, 
                                         0.31164502672127237, 0.39706458757579027, -0.15933895962032174, 
                                         0.32429999999999998, 0.27649918015687308, 0.041648835918740232, 
                                         0.61219034398622429, 0.71149999999999991, 0.56440610192862939, 
                                         0.23050000000000001, 0.092600000000000002, 0.84970000000000001, 
                                         0.31730000000000008, 0.42870903490514395, 0.37990000000000007, 
                                         0.47449999999999998, 0.91790000000000005, 0.10948694612682254, 
                                         0.056420130851747859, 0.52139999999999997, 0.90210000000000001, 
                                         0.74970000000000003, 1.4916, 1.5539000000000001, 0.52080000000000004, 
                                         0.25145394427178308, 0.23882030484464734, 0.14241702215041707, 
                                         0.38809999999999995, 0.38890000000000002, 0.89941571250126162, 
                                         0.68359999999999999, 0.1028, 0.44969999999999999, 0.77045078163444181, 
                                         0.23967849111488815, 1.1845000000000001, 0.72476587236401568, 
                                         0.35088425032748499, 0.0877, 0.050608575581014691, 0.90559999999999996, 
                                         0.31259999999999999, 0.36966697950581001, 0.37580000000000002, 
                                         0.28404999999999997, 0.35115037167709656, 1.7808137905961838, 
                                         1.1536, 0.95909999999999995, 0.1699, 0.47799999999999998, 0.84844728366956212, 
                                         1.0894771934669136, 1.1186211336629401, 0.6434105409890436, 1.4265000000000001, 
                                         0.65114326578336812, 1.2911064981717997, 0.86550000000000016, 
                                         0.2838, 0.23766779528339893, 1.2539763424435058, 0.60461682236106273, 
                                         -0.12629059055923353, 0.13001042301884957, 0.38429999999999997
  ), variance = c(0.038838197765015056, 0.11615415634475244, 
                          0.033300000000000003, 0.080669970710859545, 0.063900000000000012, 
                          0.07010000000000001, 0.0023999999999999994, 0.0014000000000000002, 
                          0.00011953331955172178, 0.00036666666666666656, 9.9959999999999988e-06, 
                          0.10026613897143627, 0.008479999999999998, 0.0092999999999999992, 
                          0.020237005388548553, 0.030300000000000004, 0.0014322376285673007, 
                          0.023000000000000007, 0.0044742134009538364, 0.03725694163325375, 
                          0.0018872170365519417, 0.053999999999999999, 0.018369292964640099, 
                          0.055094610332851894, 0.0032243057813258163, 0.015400000000000002, 
                          0.05333901352733169, 0.017899999999999999, 0.05018745135162845, 
                          0.021800000000000003, 0.037439271088948019, 0.0092505687355436891, 
                          0.0029545454660167367, 0.0045999999999999999, 0.026439894100004527, 
                          0.023000000000000007, 0.001342174237479242, 0.0049999999999999992, 
                          0.098999999999999977, 0.0084000000000000012, 0.018299999999999997, 
                          0.019815068364700619, 0.022100000000000002, 0.012699999999999998, 
                          0.017354194261738246, 0.01127602888243313, 0.00016000000000000001, 
                          0.057199999999999994, 0.034430840046899905, 0.081638319463957262, 
                          0.01981687887033701, 0.027589069422310188, 0.030272311481628494, 
                          7.0015626419235575e-05, 0.069176156679063308, 0.02869393843039415, 
                          0.01943333487987511, 0.015699379818205149, 0.0033000000000000008, 
                          0.047200000000000006, 0.073699999999999988, 0.027635146046773893, 
                          0.0089922246906977456, 0.12232010290573392, 0.023200000000000002, 
                          0.040039153835340319, 0.00032000000000000003, 0.040099999999999997, 
                          0.023499999999999993, 0.027231674943570598, 0.00055690191194043847, 
                          0.0061000000000000004, 0.0030963357842475098, 0.0044627797431889523, 
                          0.00064000000000000005, 0.0482, 0.13120000000000001, 0.013299999999999996, 
                          0.014999999999999999, 0.004679401737497273, 0.013100000000000002, 
                          0.0047000000000000002, 0.086599999999999983, 0.0014000000000000002, 
                          6.6666666666666711e-05, 0.024243128640032106, 0.019766999330721985, 
                          0.040400000000000005, 0.020168631117157752, 0.031648800689630968, 
                          0.0088999999999999999, 0.021800000000000003, 0.00012886414036986779, 
                          0.034599999999999999, 0.023433088824858842, 0.00040000000000000002, 
                          0.068689884053573208, 0.14732724887900514, 0.011731569866332624, 
                          0.034178115514898778, 0.054500000000000007, 0.020163145150945291, 
                          0.15400000000000003, 0.13988261547440278, 0.025500000000000002, 
                          0.033099999999999991, 0.039189715277057405, 0.0137, 0.0011000000000000001, 
                          0.074554031855674635, 0.020039860279798961, 0.013699997376860123, 
                          0.065219821723167301, 1.3807504831432867e-05, 0.077399999999999997, 
                          0.035100000000000006, 0.039299424098612562, 0.0090399999999999977, 
                          0.067740002168319838, 0.054007852299490948, 0.36370000000000008, 
                          0.013200000000000003, 0.021099999999999997, 0.01521373414699771, 
                          0.095299999999999982, 0.0014547328986032214, 0.028756403485850405, 
                          0.021999999999999999, 0.063294270585405377, 0.051200000000000002, 
                          0.034496524341902188, 0.011866378169406149, 0.0053105033648421897, 
                          0.029099999999999994, 0.00089999999999999987, 0.019599999999999999, 
                          0.03144167085165922, 0.03450000000000001, 0.063800000000000009, 
                          0.027297923727116102, 0.041677701121741774, 0.10427266126693882, 
                          0.0063, 0.00018074611478672672, 0.0050872292539395443, 0.12215386067485151, 
                          0.021238322910013187, 0.012600000000000002, 0.01449512503576697, 
                          0.00059534663465000504, 0.034852613673020887, 0.012499999999999997, 
                          0.055628523236345692, 0.0103, 0.00080000000000000004, 0.054500000000000007, 
                          0.067500000000000018, 0.025592437073637071, 0.027499999999999993, 
                          0.0047000000000000002, 0.0137, 0.0031992029297761094, 0.0027327492856041051, 
                          0.016299999999999999, 0.11240000000000001, 0.065899999999999986, 
                          0.081499999999999989, 0.077799999999999994, 0.0339, 0.017155422997077004, 
                          0.042145594551104018, 0.049129079134956934, 0.031800000000000002, 
                          0.035800000000000005, 0.0028188737158432791, 0.070599999999999996, 
                          0.029000000000000008, 0.075899999999999995, 0.011317212449934271, 
                          0.018291980111870835, 0.2351, 0.090163128638644022, 0.045597789513047951, 
                          0.0023, 0.011363215766630375, 0.030200000000000008, 0.074999999999999997, 
                          0.012791003373210876, 0.0029333333333333325, 0.0032799999999999995, 
                          0.020809785775319278, 0.040485477345305804, 0.070300000000000015, 
                          0.029799999999999993, 0.0038000000000000013, 0.034299999999999997, 
                          0.028214625942070838, 0.047493890268555897, 0.042054040925142554, 
                          0.078727671416194647, 0.072700000000000015, 0.045678239330859391, 
                          0.077458640744070412, 0.082699999999999996, 0.039300000000000009, 
                          0.041952252413470704, 0.10226302267212048, 0.023916799228752651, 
                          0.016090664787031757, 0.0035702771184481434, 0.076899999999999982
  )), row.names = c(NA, -211L), class = c("tbl_df", "tbl", "data.frame"
  ))

library(puniform)
#> Loading 'puniform' package (Version 0.2.5)
puni_star(yi = nudge_dat$effect_size, vi = nudge_dat$variance,
          side = "right", method = "ML", boot = FALSE)
#> Error in if (get_LR_tau(prof.tau = 0, yi = yi, vi = vi, est = est, tau.est = tau.est, : missing value where TRUE/FALSE needed
```

<sup>Created on 2023-03-01 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by RobbievanAert** [comment]
Hi Lukas,

Thanks for raising this issue. You got the error, because of the very large between-study variance in your meta-analysis.  A consequence of this was that the confidence interval for the between-study variance could not be computed. 

I have changed the procedure for computing the confidence interval, and now the method works when applying it to your data. If you want to apply the method to your data, you need the development version of the package. You can install this package using this line of R code:

`devtools::install_github("RobbievanAert/puniform", ref = "develop")`

I will close this issue, but feel free to reopen it in case this is needed.

**Comment by LukasWallrich** [comment]
Very helpful, thank you! I came across this when developing a package that allows researchers to provide an interactive comparison of different analysis methods - https://github.com/LukasWallrich/metaUI. The dataset I shared is one of the examples we use. In that specific case, puniform will remain missing for now, but I am glad to hear that it is fixed and should work with a future release of your package!
